arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+		client.ListPublishersRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) ListPublishersRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers"))
+}
+
+// ListSkus gets a list of virtual machine image skus.;
+//
+func (client *VirtualMachineImagesClient) ListSkus(location string, publisherName string, offer string) (result []VirtualMachineImageResource, err error) {
+
+	req, err := client.NewListSkusRequest(location, publisherName, offer)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure preparing VirtualMachineImagesClient.ListSkus request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListSkus request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListSkus request (%v)", err)
+	}
+
+	result = []VirtualMachineImageResource{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *VirtualMachineImagesClient) NewListSkusRequest(location string, publisherName string, offer string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"location":       url.QueryEscape(location),
+		"offer":          url.QueryEscape(offer),
+		"publisherName":  url.QueryEscape(publisherName),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.ListSkusRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) ListSkusRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus"))
+}
+
+// List gets a list of virtual machine images.;
+//
+// filter is the filter to apply on the operation.
+func (client *VirtualMachineImagesClient) List(location string, publisherName string, offer string, skus string, filter string, top int, orderby string) (result []VirtualMachineImageResource, err error) {
```

Comment:
How does user do pagination on this `List` call? Assuming they don't know the total count, they're still forced to provide a `top int` argument, so I'm not sure how they can possibly discover if there is a second page or not.